packages/electron-builder/src/publish/PublishManager.ts

```diff
@@ -89,6 +90,12 @@ export class PublishManager implements PublishContext {
         }
       }
 
+      if (publishConfig.provider === "s3") {
+        const s3 = new S3({signatureVersion: "v4"})
```

Comment:
As commented above, we should do it only if bucket name contains dot — avoid net calls.